src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1609,6 +1609,13 @@ public ScrollListener(int dir, boolean block)   {
 
         /** {@inheritDoc} */
         public void actionPerformed(ActionEvent e) {
+            // If scrollbar isn't visible, stop the timer
+            if (!scrollbar.isShowing()) {
+                ((Timer)e.getSource()).stop();
+                buttonListener.handledEvent = false;
+                scrollbar.setValueIsAdjusting(false);
```

Comment:
scrollbar.setValueIsAdjusting() is actually used in the context of scroll thumb drag.
Not sure why it is also used on Ln# 1612, but no harm in resetting it to false.